File: object_relations/app/models/review.rb
class Review
 
  attr_accessor :content, :restaurant, :customer
  @@all = []

  def initialize(content, restaurant, customer)
    @content = content
    @restaurant = restaurant
    @customer = customer
    @@all << self
  end  

  def self.all
    @@all
  end

end


# Review#customer
# Review#restaurant

customer1 = Customer.new('grace')
customer2 = Customer.new('gina')

pizza = Restaurant.new('pizza')
sandwiches = Restaurant.new('sandwiches')

Review.new('it was good', pizza, customer1)
Review.new('not bad', pizza, customer2)

File: javascript/models/review.js
class Review {
  constructor(content, restaurant, customer) {
    this.content = content
    this.restaurant = restaurant
    this.customer = customer
    store.reviews.push(this)
  }

  all() {
    return store.reviews
  }

}

let todd = new Customer('todd')
let julie = new Customer('julie')

let pizza = new Restaurant('pizza')
let soup = new Restaurant('soup')

new Review('it was good', pizza, todd)
new Review('good cheese', pizza, julie)
new Review('no soup for you', soup, julie )